wordpress/wp-content/themes/webonary-zeedisplay/search.php

```diff
@@ -10,26 +10,37 @@ function openImage(image)
 
  	<div style="padding: 10px 25px;">
 	<div id="content">
-
 		<?php
-		if(strlen($_GET['s']) > 0)
+		$search_query = '';
+		$search = filter_input(INPUT_GET, 's', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+		if($search !== '')
 		{
-			$searchquery = get_search_query();
+			$search_query = get_search_query();
 		}
 		else
 		{
-			if(strlen($_GET['semdomain']))
+			$sem_domain = filter_input(INPUT_GET, 'semdomain', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+			if($sem_domain !== '')
 			{
-				$searchquery = $_GET['semdomain'];
+				$search_query = $sem_domain;
 			}
 			else
 			{
-				$searchquery = get_term($_GET['tax'], "sil_semantic_domains")->name;
+				$taxonomy = filter_input(INPUT_GET, 'tax', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+				if (get_option('useCloudBackend'))
+				{
+					$search_query = $taxonomy;
+				}
+				else
+				{
+					$term = get_term($taxonomy, 'sil_semantic_domains');
+					$search_query = isset($term) ? $term->name : $taxonomy;
+				}
 			}
 
 		}
 		?>
-		<h2 class="arh"><?php printf( __('Search results for "%s"', ZEE_LANG), $searchquery);?></h2>
+		<h2 class="arh"><?php printf( __('Search results for "%s"', ZEE_LANG), $search_query);?></h2>

```

Comment:
Yes, $search_query will be an empty string if the user hits the search button without specifying anything in the search text box. We should probably catch that with javascript validation on the search form. I had initialized $search_query to an empty string to avoid PHP warning.  Since searching for an empty string will return an empty post, the complete message that will appear back to the user is as below, which should suffice in informing the user that they specified a search for nothing.
<img width="881" alt="Screen Shot 2020-06-14 at 9 37 53 PM" src="https://user-images.githubusercontent.com/60716623/84612898-97463480-ae87-11ea-979d-584ae7b65545.png">